### ISSUE TITLE
Review where Mailpile creates temporary files

### ISSUE THREAD

**Issue by BjarniRunar** [mailpile/Mailpile#2218]
An issue I discovered while researching #2217; Mailpile creates temporary files for a few things, generally using a subdirectory within the Mailpile data directory.

Since the Mailpile data directory is likely to be on a real drive (SSD, spinning rust, or USB storage on things like the Raspberry Pi), performance here is likely to be suboptimal, not to mention unnecessary wear-and-tear on the underlying storage. This probably explains reports I've gotten that Mailpile slows down significantly on some setups and upgrading to faster storage for the Mailpile home directory helps.

Most Linux distros mount `/tmp` as a tmpfs in order to alleviate this sort of thing, and we should play nice and reconsider which data really needs to be in Mailpile's home folder (probably only potentially very large files) and which data can be spooled to the system's preferred `/tmp`.

**Comment by mtlambert** [comment]
Hi, would you mind if I gave this a try?

**Comment by BjarniRunar** [comment]
Hey @mtlambert, sorry about the late response, I haven't been paying good attention.

I came here to report that I've complete this review, and things were not as silly as I feared. Tempfiles are indeed created outside of `/tmp/` in many cases, but always because we then intend to rename/move them into place within the user's data directory. In such cases using `/tmp/` would not provide any benefit and would actually slow things down.
